modeltranslation/tests/models.py

```diff
@@ -1,16 +1,22 @@
 from django.contrib.auth.models import Permission
 from django.core import validators
 from django.db import models
-from django.utils.translation import gettext_lazy
+from django.utils.translation import gettext_lazy, get_language
 
 from modeltranslation.manager import MultilingualManager
 
 
+def get_dynamic_default():
+    lang = get_language()
+    return f"dynamic: {lang}"
+
+
 class TestModel(models.Model):
     title = models.CharField(gettext_lazy('title'), max_length=255)
     text = models.TextField(blank=True, null=True)
     url = models.URLField(blank=True, null=True)
     email = models.EmailField(blank=True, null=True)
+    dynamic_default = models.CharField(default=get_dynamic_default, max_length=255)
```

Comment:
suggestion,open to discussion:

Maybe it's better to test it with translation instead of custom function?

```suggestion
    dynamic_default = models.CharField(default=_("Hello"), max_length=255)
```

That would require adding locales to test app, but will be exactly what we need to check. Or we can use preexisting traslations from django, we're have `django.contrib.auth` app installed in the settings, so we can use any string from `contrib/auth/locale/en/LC_MESSAGES/django.po`

For example, `_("password")`

Reply:
Patch is updated, closing this conversation. I didn't think this approach was possible but I'm very happy that it is, as it leads to an even cleaner API for users of the library!